tests/ui/lint/non-local-defs/convoluted-locals-131474-with-mods.rs

```diff
@@ -42,4 +42,28 @@ const _: () = {
     }
 };
 
+// https://github.com/rust-lang/rust/issues/132427
+mod auth {
+    const _: () = {
+        pub enum ArbitraryContext {}
+
+        const _: () = {
+            impl ArbitraryContext {}
+        };
+    };
+}
+
+// https://github.com/rust-lang/rust/issues/132427
+const _: () = {
+    mod auth {
+        const _: () = {
+            pub enum ArbitraryContext {}
+
+            const _: () = {
+                impl ArbitraryContext {}
+            };
+        };
+    }
+};
```

Comment:
Question: ... just to double-check: does this logic also work if there's deeper nesting or mixed interleavings between layers of `mod`s and anon consts `const _: ()`? Like

```rs
mod a {
    mod b {
        const _: () = {
            pub enum ArbitraryContext {}

            const _: () = {
                impl ArbitraryContext {}
            };
        };
    }
}
```

or

```rs
mod foo {
    const _: () = {
        mod auth {
            const _: () = {
                pub enum ArbitraryContext {}

                const _: () = {
                    impl ArbitraryContext {}
                };
            };
        }
    };
}
```

Now, usually one does not simply write such code, but alas.

Reply:
It should, let me check and add tests for those as well.

Reply:
And it indeed does. Extra tests added.